Reference/Searching/Examine/transforming-index-values.md

```diff
@@ -0,0 +1,120 @@
+
+Instead you can use **TransformingIndexValues** (used to be called GatheringNodeData in Umbraco 7). TransformingIndexValues is an Examine event where a new field is created in the index that contains the data from several other fields, you can then search on just that field. This is done via a [composer](../../../Implementation/Composing/index.md).
+
+## Creating a TransformingIndexValues event
+
+:::note
+This example will build upon the Umbraco Starterkit as it is a good starting point for some content. Feel free to use your own site, but some content may need to be generated.
+:::
+
+So to add a TransformingIndexValues event we will add a controller that inherits from `IComponent`. Something like this:
+
+```cs
+public class ExamineEvents : IComponent
+{
+    private readonly IExamineManager _examineManager;
+    private readonly ILogger _logger;
+
+    public ExamineEvents(IExamineManager examineManager, ILogger logger)
+    {
+        _examineManager = examineManager;
+        _logger = logger;
+    }
+
+    public void Initialize()
+    {
+        if (!_examineManager.TryGetIndex("ExternalIndex", out IIndex index))
+            throw new InvalidOperationException("No index found by name ExternalIndex");
+
+        //we need to cast because BaseIndexProvider contains the TransformingIndexValues event
+        if (!(index is BaseIndexProvider indexProvider))
+            throw new InvalidOperationException("Could not cast");
+
+        indexProvider.TransformingIndexValues += IndexProviderTransformingIndexValues;
+    }
+
+    private void IndexProviderTransformingIndexValues(object sender, IndexingItemEventArgs e)
+    {
+        //todo
+    }
+
+    public void Terminate()
+    {
+    }
+}
+```
+
+If this syntax is confusing you can read more about [Components](../../../Implementation/Composing/index.md). We can now add the logic to combine fields in the `IndexProviderTransformingIndexValues` method:
+
+```cs
+if (e.ValueSet.Category == IndexTypes.Content)
+{
+    try
+    {
+        var combinedFields = new StringBuilder();
+        foreach (var fieldValues in e.ValueSet.Values)
+        {
+            foreach (var value in fieldValues.Value)
+            {
+                if (value != null)
+                {
+                    combinedFields.AppendLine(value.ToString());
+                }
+            }            
+        }
+
+        e.ValueSet.Add("combinedField", combinedFields.ToString());
+    }
+    catch (Exception ex)
+    {
+        _logger.Error<ExamineEvents>(ex, "Error combining fields for {ValueSetId}", e.ValueSet.Id);
+    }
+}
+```
+
+So at this point we have done something along the lines of:
+- Get the ExternalIndex
+- Get the valueset for the ExternalIndex
+- For each field, add the content to a new field called `combinedField`
+
+Before this works the component will have to be registered in a composer. If you already have a composer you can add it to that one, for this example we will make a new composer:
```

Comment:
missing the word `but`, so should be `but for this example we will make a new composer:`